engines/osxnative/sandbox.go

```diff
@@ -162,19 +162,19 @@ func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
 			}
 		}()
 
-		userInfo, err := osuser.Lookup(u.name)
-		if err != nil {
+		userInfo, err2 := osuser.Lookup(u.name)
+		if err2 != nil {
 			s.context.LogError("Error looking up for user \""+u.name+"\": ", err, "\n")
 		} else {
-			uid, err := strconv.ParseUint(userInfo.Uid, 10, 32)
-			if err != nil {
-				s.context.LogError("ParseUint failed to convert ", userInfo.Uid, ": ", err, "\n")
+			uid, err2 := strconv.ParseUint(userInfo.Uid, 10, 32)
+			if err2 != nil {
+				s.context.LogError("ParseUint failed to convert ", userInfo.Uid, ": ", err2, "\n")
 				return nil, engines.ErrNonFatalInternalError
 			}
 
-			gid, err := strconv.ParseUint(userInfo.Gid, 10, 32)
-			if err != nil {
-				s.context.LogError("ParseUint failed to convert ", userInfo.Gid, ": ", err, "\n")
+			gid, err2 := strconv.ParseUint(userInfo.Gid, 10, 32)
+			if err2 != nil {
+				s.context.LogError("ParseUint failed to convert ", userInfo.Gid, ": ", err2, "\n")
```

Comment:
Hmm... Fixed it, but this seems like a rather fragile control flow... We should probably rethink that.

Perhaps set a boolean `cleanup = true` or something... to trigger the deferred function.. Or even better not use defer like this.. -- I'm not immediately sure how to structure this an elegant manner.